libraries/crypto/src/ecdsa.rs

```diff
@@ -38,6 +41,7 @@ pub struct Signature {
     s: NonZeroExponentP256,
 }
 
+#[cfg_attr(feature = "derive_debug", derive(Clone))]
```

Comment:
I think we noticed that at least guarding `Debug` is important to reduce the binary size significantly. I didn't check if this is also true for `Clone` etc, but I'd say we should keep them?

Reply:
Who noticed and how did they notice? If the binary size is bigger when adding dead-code, it should be reported to the compiler/linker.

Here is how I test it:

```
% git rev-parse HEAD
deeabe026f07e714a118ab365914b5f9b7d6de34
% ./deploy.py --board=nrf52840dk --opensk --panic-console --programmer=none
[... omitted ...]
% stat -c%s target/thumbv7em-none-eabi/release/ctap2                       
411632
% edit Cargo.toml
- panic_console = ["lang_items/panic_console"]
+ panic_console = ["lang_items/panic_console", "crypto/derive_debug"]
% ./deploy.py --board=nrf52840dk --opensk --panic-console --programmer=none
[... omitted ...]
% stat -c%s target/thumbv7em-none-eabi/release/ctap2                       
411448
```

The size is actually smaller, not sure why. But I would not be surprised if trying to manually optimize stuff gives worse results than letting the compiler do its job.

Reply:
I think that was @gendx , i.e. see his comment in [libtock](https://github.com/google/OpenSK/blob/67a94c8067f12e1c642ee338b81722495240659c/third_party/libtock-drivers/src/result.rs#L7). The discussion is pretty old I think, so maybe we have to re-evaluate?

This feels like it's worth a new issue / PR if we generally move to no `cfg` for `derive`?

Reply:
Yes, this question is orthogonal to the PR. I'll create a separate issue.

Also, the comment you point out is unrelated to what I'm saying. That comment makes sense, because it's about avoiding the need of error types implementing the `Debug` trait. What I'm saying is that we don't need to feature-gate generated dead-code, like deriving the `Debug` trait.

Reply:
#251 to continue this discussion.